bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/AbstractConfiguration.java

```diff
@@ -310,6 +312,60 @@ public String getLedgerManagerType() {
         return getString(LEDGER_MANAGER_TYPE);
     }
 
+    /**
+     * Set the flag to allow using shaded ledger manager factory class for
+     * instantiating a ledger manager factory.
+     *
+     * @param allowed
+     *          the flag to allow/disallow using shaded ledger manager factory class
+     * @return configuration instance.
+     */
+    public T setAllowShadedLedgerManagerFactoryClass(boolean allowed) {
```

Comment:
I don't think this is needed. If you set "ShadedLedgerManagerFactoryClassPrefix" it implies you want to allow it.

Reply:
would prefer having an explicit flag on controlling this

Reply:
Sure. My issue with it is that it exposes it in a public api, so it's hard to change later. Not super strongly opposed though.